src/NuGet.Clients/NuGet.SolutionRestoreManager/IInfoBarService.cs

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.SolutionRestoreManager
+{
+    public interface IInfoBarService
```

Comment:
I know we had done some of this together, but if you do want to truly follow OOP rules, you can hide that this showing an info bar from restore. 

Restore is basically telling a component that there are packages with vulnerabilities within the sln. The fact that it's info bar is irrelevant to it. 

Don't feel too strongly, but think it's a good diea to tweak it. @NuGet/nuget-client might have some ideas